docs/reference/ingest/processors/registered-domain.asciidoc

```diff
@@ -0,0 +1,83 @@
+[role="xpack"]
+[testenv="basic"]
+[[registered-domain-processor]]
+=== Registered domain processor
+++++
+<titleabbrev>Registered Domain</titleabbrev>
+++++
+
+Extracts the registered domain (also known as effective top-level domain or
+eTLD), sub-domain, and top-level domain from a fully qualified domain name
+(FQDN). Uses the registered domains defined in the
+https://publicsuffix.org/[Mozilla Public Suffix List].
+
+[[registered-domain-options]]
+.Registered Domain Options
+[options="header"]
+|======
+| Name               | Required | Default | Description
+| `field`            | yes      |         | Field containing the source FQDN.
+| `target_field`     | no       | `<empty string>` | Field under which the
+extracted domain components will be written. If the empty string, the domain
+components will be added to the root of the document.
+| `ignore_missing`   | no       | `true`  | If `true` and any required fields
+are missing, the processor quietly exits without modifying the document.
+
+include::common-options.asciidoc[]
+|======
+
+[discrete]
+[[registered-domain-processor-ex]]
+===== Examples
+
+The following example illustrates the use of the registered domain processor:
+
+[source,console]
+----
+POST _ingest/pipeline/_simulate
+{
+  "pipeline": {
+    "processors": [
+      {
+        "registered_domain": {
+          "field": "fqdn",
+          "target_field": "url"
+        }
+      }
+    ]
+  },
+  "docs": [
+    {
+      "_source": {
+        "fqdn": "www.amazon.co.uk"
```

Comment:
I'd use `example.com` or `example.org` in any docs or examples.
```suggestion
        "fqdn": "www.example.org"
```

Reply:
Good point. I did want something with a two-part TLD to illustrate that it's not just splitting the FQDN on periods. Would something such as `www.example.co.uk` work?

Reply:
I'd use `www.example.ac.uk`. `ac.uk` is reserved for academic use and would be harder to use for nefarious means in the future.